File: making_decisions/python/numbers_to_names.py
# Pratice 21. Numbers to names
# Output:
#   Please enter the number of the month: 3
#   The name of the month is March.
# Constraint:
#   - Use a switch or case statement for this program.
#   - Use a single output statement for this program.

#!/usr/bin/env python

import sys

def input_process(in_question):
    return input(in_question) if sys.version_info >= (3,0) else raw_input(in_question)

def switch_state(in_option):
    return {
        1 : 'January',
        2 : 'February',
        3 : 'March',
        4 : 'April',
        5 : 'May',
        6 : 'June',
        7 : 'July',
        8 : 'August',
        9 : 'September',
        10 : 'October',
        11 : 'November',
        12 : 'December'
    }.get(in_option)

if __name__ == '__main__':
    try:
        number_month = int(input_process('Please enter the number of the month: '))
    except:
        print ('You must input only numbers.')
    else:
        if number_month < 1 or number_month > 12:
            print ('Wrong input value, you must input number of the month.')
            exit()

        print ('The name of the month is %s.' % switch_state(number_month))

File: making_decisions/go/numbers_to_names.go
/*
 * Pratice 21. Numbers to names
 * Output:
 *   Please enter the number of the month: 3
 *   The name of the month is March.
 * Constraint:
 *   - Use a switch or case statement for this program.
 *   - Use a single output statement for this program.
 */

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func get_month_str(number_month int) string {
	switch {
	case number_month == 1:
		return "January"
	case number_month == 2:
		return "February"
	case number_month == 3:
		return "March"
	case number_month == 4:
		return "April"
	case number_month == 5:
		return "May"
	case number_month == 6:
		return "June"
	case number_month == 7:
		return "July"
	case number_month == 8:
		return "August"
	case number_month == 9:
		return "September"
	case number_month == 10:
		return "October"
	case number_month == 11:
		return "November"
	case number_month == 12:
		return "December"
	}
	return ""
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Println("Please enter the number of the month: ")
	number_month_str, _ := reader.ReadString('\n')

	number_month, _ := strconv.Atoi(strings.TrimSpace(number_month_str))

	if number_month < 1 || number_month > 12 {
		fmt.Println("Wrong input value, you must input number of the month.")
		return
	}

	fmt.Printf("The name of the month is %s.\n", get_month_str(number_month))
}
